src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/PostBuildEventValueProviderTests.cs

```diff
@@ -0,0 +1,524 @@
+
+using Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties;
+using Microsoft.VisualStudio.ProjectSystem.VS.Utilities;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ProjectSystemTrait]
+    public class PostBuildEventValueProviderTests
+    {
+        private static PostBuildEventValueProvider.PostBuildEventHelper systemUnderTest =
+            new PostBuildEventValueProvider.PostBuildEventHelper();
+
+        [Fact]
+        public static void GetPropertyTest_AllTargetsPresent()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <Target Name=""PreBuild"" BeforeTargets=""PreBuildEvent"">
+    <Exec Command=""echo &quot;prebuild output&quot;"" />
+  </Target>
+
+  <Target Name=""PostBuild"" AfterTargets=""PostBuildEvent"">
+    <Exec Command=""echo &quot;post build output&quot;"" />
+  </Target>
+
+</Project>
+".AsProjectRootElement();
+            var actual = systemUnderTest.GetProperty(root);
+            Assert.Equal(@"echo ""post build output""", actual);
+        }
+
+        [Fact]
+        public static void GetPropertyTest_PostBuildTargetPresent()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <Target Name=""PostBuild"" AfterTargets=""PostBuildEvent"">
+    <Exec Command=""echo &quot;post build output&quot;"" />
+  </Target>
+
+</Project>
+".AsProjectRootElement();
+            var actual = systemUnderTest.GetProperty(root);
+            Assert.Equal(@"echo ""post build output""", actual);
+        }
+
+        [Fact]
+        public static void GetPropertyTest_NoTargetsPresent()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+</Project>
+".AsProjectRootElement();
+            var actual = systemUnderTest.GetProperty(root);
+            Assert.Null(actual);
+        }
+
+        [Fact]
+        public static void GetPropertyTest_ExistingProperties()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+    <PostBuildEvent>echo $(ProjectDir)</PostBuildEvent>
+  </PropertyGroup>
+
+</Project>
+".AsProjectRootElement();
+            var actual = systemUnderTest.GetProperty(root);
+            Assert.Equal(@"echo $(ProjectDir)", actual);
+        }
+
+        [Fact]
+        public static void SetPropertyTest_NoTargetsPresent()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
```

Comment:
`root.Save(TextWriter)` and `root.Save(string path)` use completely different code paths.  I'll change the tests to write out a file so the newlines are not changed